Add glofas

mnlevy1981 opened the pull request:
### Description of changes
A companion to https://github.com/ESMCI/ccs_config_cesm/pull/255, this PR is necessary to support a new runoff data set

### Specific notes

Contributors other than yourself, if any: @manishvenu


CDEPS Issues Fixed (include github issue #): N/A

Are there dependencies on other component PRs (if so list): CESM will also need https://github.com/ESMCI/ccs_config_cesm/pull/255

Are changes expected to change answers (bfb, different to roundoff, more substantial): this should be bfb in all tested configurations; we are adding a new data stream, but not altering any existing ones.

Any User Interface Changes (namelist or namelist defaults changes): No

Testing performed (e.g. aux_cdeps, CESM prealpha, etc): This will be thoroughly tested before being taken out of draft mode

Hashes used for testing:

mnlevy1981 committed:
Initial changes for CROCODILE

Was based on cesm2_3_alpha17f, but diffs applied to cesm3_0_alpha03a

mnlevy1981 committed:
Merge tag 'cdeps1.0.57' into crocodile

cdeps1.0.57

manishvenu committed:
Glofas streamentry

mnlevy1981 committed:
Merge remote-tracking branch 'crocodile/runoff' into add_glofas

mnlevy1981 committed:
Consistent ordering of modifiers

I also removed a rof.globas entry that was actually on the wrong grid

mnlevy1981 committed:
Update to most recent glofas

mnlevy1981 commented:
@manishvenu this should come out of draft and be marked ready for review, right? Or does https://github.com/ESMCI/ccs_config_cesm/pull/255 need to be finished first?

manishvenu commented:
> @manishvenu this should come out of draft and be marked ready for review, right? Or does [ESMCI/ccs_config_cesm#255](https://github.com/ESMCI/ccs_config_cesm/pull/255) need to be finished first?

Hey @mnlevy1981, I think this one can be done first? I'm not an expert on what order it needs to come in, but this one seems right. 

I think it's ready for review, but will post a new GLOFAS path tomorrow.

manishvenu commented:
Hey @mnlevy1981,

I moved the GLOFAS files to a similar place to the JRA runoff files, here is the new paths and command:

Files: $DIN_LOC_ROOT/ocn/glofas/v4/GLOFAS.v4.dis24.%y.nc
MeshFIle: $DIN_LOC_ROOT/ocn/glofas/v4/metadata/GLOFAS.v4.ESMFmesh.nc

The one question I had is that all the runoff meshfiles are in dlnd. Do we want to move GLOFAS meshfile there?

Thanks,
Manish V.

mnlevy1981 commented:
> The one question I had is that all the runoff meshfiles are in dlnd. Do we want to move GLOFAS meshfile there?

Yeah, I think we want it to be named `lnd/dlnd7/GLOFAS/GLOFAS.v4.dis24.ESMFmesh_20260501.nc` (and we need to drop the `NaN` `_FillValue` attributes)

mnlevy1981 committed:
Merge branch 'main' into add_glofas

mnlevy1981 committed:
Update file locations

The glofas mesh file is now in lnd/dlnd7, and the runoff data files were moved
to ocn/glofas

manishvenu commented:
Hey @mnlevy1981,

I don't have permissions to add the ESMF Mesh here. I've generated the mesh here: /glade/campaign/cesm/cesmdata/cseg/inputdata/ocn/glofas/v4/metadata/GLOFAS.v4.dis24.ESMFmesh_20260501.nc

mnlevy1981 commented:
> Hey @mnlevy1981,
> 
> I don't have permissions to add the ESMF Mesh here. I've generated the mesh here: /glade/campaign/cesm/cesmdata/cseg/inputdata/ocn/glofas/v4/metadata/GLOFAS.v4.dis24.ESMFmesh_20260501.nc

I just copied it to the correct directory, and we can run `rimport` next week after testing everything in a fresh sandbox

billsacks committed:
Merge branch 'main' into add_glofas
